### PR TITLE
Switch to Vite

### PR DESCRIPTION
**Pull request by tim-macphail**
# Switch to vite from webpack for improved performance in development and build

## Limitations:
- Some of the kinks are still being worked out like I had to put most of `tailwind.config.cjs` into `Utilities.tsx` https://github.com/techstartucalgary/lifeline/blob/ac5a128339d91d7453adaa7b731cd6904ed37fdc/client/src/Utilities.tsx#L9 @cloudyyoung maybe you can help with this
- Prettier is not configured (do we need it with eslint?)
- Probably some other things/files slipped through the cracks in the move over

### CODE REVIEW

File: client/.eslintignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+dist/
+build/
```
Comment (cloudyyoung) [inline client/.eslintignore:2]: This should live in the root of repository. Or this file can be gone if you apply the next change.

File: client/vite.config.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { defineConfig } from "vite";
+import react from "@vitejs/plugin-react-swc";
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [react()],
```
Comment (cloudyyoung) [inline client/vite.config.ts:6]:
```suggestion
  plugins: [react()],
  build: {
    outDir: '../build'
  }
```

**Review by cloudyyoung — COMMENTED** [review]
Since Vite doesn't perform typecheck in compiling, could add a VSCode extension recommendation to the project.

```
Name: JavaScript and TypeScript Nightly
Id: ms-vscode.vscode-typescript-next
Description: Enables typescript@next to power VS Code's built-in JavaScript and TypeScript support
Version: 5.0.202302070
Publisher: Microsoft
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next
```


And if I'm understand this correctly, `Utilities.jsx` contains a hard copy of `tailwind.config.js`?

**Comment by tim-macphail** [comment]
> Since Vite doesn't perform typecheck in compiling, could add a VSCode extension recommendation to the project.
> 
> ```
> Name: JavaScript and TypeScript Nightly
> Id: ms-vscode.vscode-typescript-next
> Description: Enables typescript@next to power VS Code's built-in JavaScript and TypeScript support
> Version: 5.0.202302070
> Publisher: Microsoft
> VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next
> ```
> 
> And if I'm understand this correctly, `Utilities.jsx` contains a hard copy of `tailwind.config.js`?

I added recommended extensions and settings. And yes, I put a hard copy into `Utilities.tsx` so that I could get it to compile and havent really tried to fix it yet

**Comment by tim-macphail** [comment]
> Since Vite doesn't perform typecheck in compiling, could add a VSCode extension recommendation to the project.
> 
> ```
> Name: JavaScript and TypeScript Nightly
> Id: ms-vscode.vscode-typescript-next
> Description: Enables typescript@next to power VS Code's built-in JavaScript and TypeScript support
> Version: 5.0.202302070
> Publisher: Microsoft
> VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next
> ```
> 
> And if I'm understand this correctly, `Utilities.jsx` contains a hard copy of `tailwind.config.js`?

Yeah i don't know how to deal with the projectConfig

**Pull request merged**